### DISCUSSION TITLE
small eps image resized to jpg, but it appears blurry

### DISCUSSION THREAD

**Discussion by taomin597715379** [ImageMagick/ImageMagick#2122] [Help]
```
convert -colorspace sRGB -density 200 -strip -trim ../data/eps/908754658291351603.eps -resize 1280x769 abc.jpg
```
small eps:
[abc.zip](https://github.com/ImageMagick/ImageMagick/files/4749714/abc.zip)

After using the convert command to resized, the picture becomes blurred, but It is normal to export using the AI (Adobe Illustrator) tool, i don't know why ?

**Comment by fmw42** [comment]
Your EPS file contains a raster image. So it is not like resizing a vector file. So resizing by large factors will likely blur some. The best you can do is use a larger density and resize (down) to your desired size.  Also best to move the trim to after reading the EPS. So try

`convert -colorspace sRGB -density 600 -strip 908754658291351603.eps -trim -resize 1280x769 abc.jpg`

![abc](https://user-images.githubusercontent.com/13458218/84105467-4210ab80-a9cd-11ea-939a-b7eed63b6505.jpg)

**Comment by taomin597715379** [comment]
I Got it，so how to judge whether an EPS contains a raster image?

**Reply by fmw42** [reply]
One method is to convert it to PDF. For example, I open it in my Mac Preview and it automatically converts to PDF. I save it  and then run pdfimages.

pdfimages 908754658291351603.pdf xxx

That save any imbedded raster image in ppm format or optionally PNG or TIFF as desired.



[xxx-000.ppm.zip](https://github.com/ImageMagick/ImageMagick/files/4753737/xxx-000.ppm.zip)
